matrix_sdk/src/request_builder.rs

```diff
@@ -405,6 +406,92 @@ impl Into<register::Request> for RegistrationBuilder {
     }
 }
 
+/// Create a builder for making get_message_event requests.
+///
+/// # Examples
+/// ```
+/// # use std::convert::TryFrom;
+/// # use matrix_sdk::{Client, RoomSearchBuilder};
+/// # use matrix_sdk::api::r0::directory::get_public_rooms_filtered::{self, RoomNetwork, Filter};
+/// # use url::Url;
+/// # let homeserver = Url::parse("http://example.com").unwrap();
+/// # let mut rt = tokio::runtime::Runtime::new().unwrap();
+/// # rt.block_on(async {
+/// # let last_sync_token = "".to_string();
+/// let mut client = Client::new(homeserver).unwrap();
+///
+/// let generic_search_term = Some("matrix-rust-sdk".to_string());
+/// let mut builder = RoomSearchBuilder::new();
+/// builder
+///     .filter(Filter { generic_search_term, })
+///     .since(last_sync_token)
+///     .room_network(RoomNetwork::Matrix);
+///
+/// client.get_public_rooms_filtered(builder).await.is_err();
+/// # })
+/// ```
+#[derive(Clone, Debug, Default)]
+pub struct RoomSearchBuilder {
+    server: Option<String>,
+    limit: Option<UInt>,
+    since: Option<String>,
+    filter: Option<Filter>,
+    room_network: Option<RoomNetwork>,
+}
+
+impl RoomSearchBuilder {
+    /// Create a `RoomSearchBuilder` builder to make a `get_message_events::Request`.
+    ///
+    /// The `room_id` and `from`` fields **need to be set** to create the request.
```

Comment:
If there are fields that need to be set let's put them in the constructor so you can't build an invalid request. 

That being said that docstring is for the `MessagesRequestBuilder` and I realize that i failed to mention this for that builder.